.gitignore

```diff
@@ -5,7 +5,7 @@ node_modules/
 *.egg-info/
 .ipynb_checkpoints
 *.tsbuildinfo
-jupyterlite_pyodide_kernel/labextension
+src/jupyterlite_pyodide_kernel/labextension
```

Comment:
Do we need to move this folder in this PR? Are we expected more Python packages to be added to the repo later?

Would need to check how the releaser handles that. It probably breaks it without extra config.

Reply:
It would be nice to keep `jupyterlite_pyodider_kernel` to the top-level for now until we really need to move it down to a nested `src` folder or similar.

Reply:
The `src` layout is widely used in a lot of packages, but i'll grant i haven't used it with the releaser. 

This avoids the possibility of it shadowing the as-installed package in `site-packages`. For some of the coverage stuff to work, it took weird junk like deleting the source and forcing a non-`-e .` install. 

This layout is supported by the all the build backends at this point, and releaser probably needs to be updated, if it can't yet do that.

It is weird that PEP 621 doesn't have a canonical place that information, but that's the brave new world.